openwisp_notifications/handlers.py

```diff
@@ -274,16 +274,58 @@ def notification_type_registered_unregistered_handler(sender, **kwargs):
 
 
 @receiver(
-    pre_save,
+    post_save,
     sender=OrganizationUser,
     dispatch_uid='create_orguser_notification_setting',
 )
-def notification_setting_org_user_created(instance, **kwargs):
-    if instance.is_admin:
-        tasks.ns_organization_user_added_or_updated.delay(
-            instance_id=instance.pk,
-            instance_user_id=instance.user_id,
-            instance_org_id=instance.organization_id,
+def notification_setting_org_user_created(instance, created, **kwargs):
+    if created and instance.is_admin:
```

Comment:
what if the OrganizationUser is changed and is_admin is set from False to True? Then notification settings wouldn't be created right?
Is that solved with the new receiver below?

Reply:
Yes, the `pre_save` receiver takes care of that.